### ISSUE TITLE
Style/RedundantReturn throws exception on case statement

### ISSUE THREAD

**Issue by ptarjan** [bbatsov/rubocop#3649]
On this file: 

```
$ cat a.rb
def bar
  case a
  when 1
  else
    'oops'
  end
end
```

It fails with this backtrace: 

```
An error occurred while Style/RedundantReturn cop was inspecting /private/tmp/a.rb.
undefined method `type' for nil:NilClass
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/cop/style/redundant_return.rb:79:in `check_branch'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/cop/style/redundant_return.rb:102:in `check_when_node'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/cop/style/redundant_return.rb:96:in `block in check_case_node'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/cop/style/redundant_return.rb:96:in `each'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/cop/style/redundant_return.rb:96:in `check_case_node'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/cop/style/redundant_return.rb:81:in `check_branch'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/cop/style/redundant_return.rb:75:in `on_method_def'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/cop/mixin/on_method_def.rb:14:in `on_defs'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/cop/commissioner.rb:41:in `block (2 levels) in on_defs'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/cop/commissioner.rb:96:in `with_cop_error_handling'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/cop/commissioner.rb:40:in `block in on_defs'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/cop/commissioner.rb:39:in `each'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/cop/commissioner.rb:39:in `on_defs'
(eval):2:in `block in on_begin'
(eval):2:in `each'
(eval):2:in `on_begin'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/cop/commissioner.rb:45:in `on_begin'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/ast_node/traversal.rb:141:in `on_while'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/cop/commissioner.rb:45:in `on_module'
(eval):2:in `block in on_begin'
(eval):2:in `each'
(eval):2:in `on_begin'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/cop/commissioner.rb:45:in `on_begin'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/ast_node/traversal.rb:12:in `walk'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/cop/commissioner.rb:58:in `investigate'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/cop/team.rb:120:in `investigate'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/cop/team.rb:108:in `offenses'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/cop/team.rb:51:in `inspect_file'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/runner.rb:243:in `inspect_file'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/runner.rb:190:in `block in do_inspection_loop'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/runner.rb:222:in `block in iterate_until_no_changes'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/runner.rb:215:in `loop'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/runner.rb:215:in `iterate_until_no_changes'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/runner.rb:186:in `do_inspection_loop'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/runner.rb:101:in `block in file_offenses'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/runner.rb:111:in `file_offense_cache'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/runner.rb:99:in `file_offenses'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/runner.rb:90:in `process_file'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/runner.rb:68:in `block in each_inspected_file'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/runner.rb:65:in `each'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/runner.rb:65:in `reduce'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/runner.rb:65:in `each_inspected_file'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/runner.rb:57:in `inspect_files'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/runner.rb:36:in `run'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/cli.rb:71:in `execute_runner'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/lib/rubocop/cli.rb:27:in `run'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/bin/rubocop:13:in `block in <top (required)>'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
/Users/pt/.rbenv/versions/2.2.5/lib/ruby/gems/2.2.0/gems/rubocop-0.44.1/bin/rubocop:12:in `<top (required)>'
/Users/pt/.rbenv/versions/2.2/bin/rubocop:23:in `load'
/Users/pt/.rbenv/versions/2.2/bin/rubocop:23:in `<main>'
.
1 file inspected, no offenses detected

1 error occurred:
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
Mention the following information in the issue report:
0.44.1 (using Parser 2.3.1.4, running on ruby 2.2.5 x86_64-darwin15)
Finished in 0.1586333679999825 seconds
```

**Comment by Drenmi** [comment]
Pretty sure this was fixed in #3645, as I can no longer reproduce on current `master` using your example. 😀

**Comment by ptarjan** [comment]
Why wasn't a point release cut for it? It is failing all over our codebase

**Comment by shepmaster** [comment]
This also extends to empty `if` statements:

```ruby
def init
  if true
  else
    puts "hi"
  end
end
```

`0.44.1 (using Parser 2.3.1.4, running on ruby 2.3.1 x86_64-darwin16)`

**Comment by jonas054** [comment]
Fixed in #3621 and released in `v0.45.0`.
